src/main/interface/smartaudio_protocol.c

```diff
@@ -0,0 +1,187 @@
+    }
+}
+
+static void smartaudioUnpackSettings(smartaudioSettings_t *settings, const smartaudioSettingsResponseFrame_t *frame)
+{
+    settings->channel = frame->channel;
+    settings->power = frame->power;
+    smartaudioUnpackFrequency(settings, frame->frequency);
+    smartaudioUnpackOperationMode(settings, frame->operationMode, true);
+}
+
+void smartaudioFrameGetSettings(smartaudioCommandOnlyFrame_t *frame)
+{
+    smartaudioFrameInit(SMARTAUDIO_CMD_GET_SETTINGS, &frame->header, 0);
+    smartaudioCRC(&frame->crc, frame, sizeof(smartaudioFrameHeader_t));
+}
+
+void smartaudioFrameGetPitmodeFrequency(smartaudioU16Frame_t *frame)
+{
+    smartaudioFrameInit(SMARTAUDIO_CMD_SET_FREQUENCY, &frame->header, sizeof(frame->payload));
+    frame->payload = 0x4000;
+    const uint8_t frameLength = sizeof(smartaudioFrameHeader_t) + sizeof(frame->payload);
+    smartaudioCRC(&frame->crc, frame, frameLength);
+}
+
+void smartaudioFrameSetPower(smartaudioU8Frame_t *frame, const uint8_t power)
+{
+    smartaudioFrameInit(SMARTAUDIO_CMD_SET_POWER, &frame->header, sizeof(frame->payload));
+    frame->payload = power;
+    const uint8_t frameLength = sizeof(smartaudioFrameHeader_t) + sizeof(frame->payload);
+    smartaudioCRC(&frame->crc, frame, frameLength);
+}
+
+void smartaudioFrameSetBandChannel(smartaudioU8Frame_t *frame, const uint8_t band, const uint8_t channel)
+{
+    smartaudioFrameInit(SMARTAUDIO_CMD_SET_CHANNEL, &frame->header, sizeof(frame->payload));
+    frame->payload = SMARTAUDIO_BANDCHAN_TO_INDEX(band, channel);
+    const uint8_t frameLength = sizeof(smartaudioFrameHeader_t) + sizeof(frame->payload);
+    smartaudioCRC(&frame->crc, frame, frameLength);
+}
+
+void smartaudioFrameSetFrequency(smartaudioU16Frame_t *frame, const uint16_t frequency, const bool pitmodeFrequency)
+{
+    smartaudioFrameInit(SMARTAUDIO_CMD_SET_FREQUENCY, &frame->header, sizeof(frame->payload));
+    frame->payload = frequency | (pitmodeFrequency ? 0x8000 : 0x00);
+    const uint8_t frameLength = sizeof(smartaudioFrameHeader_t) + sizeof(frame->payload);
+    smartaudioCRC(&frame->crc, frame, frameLength);
+}
+
+void smartaudioFrameSetOperationMode(smartaudioU8Frame_t *frame, const uint8_t operationMode)
+{
+    smartaudioFrameInit(SMARTAUDIO_CMD_SET_MODE, &frame->header, sizeof(frame->payload));
+    frame->payload = operationMode;
+    const uint8_t frameLength = sizeof(smartaudioFrameHeader_t) + sizeof(frame->payload);
+    smartaudioCRC(&frame->crc, frame, frameLength);
+}
+
+bool smartaudioParseResponseBuffer(smartaudioSettings_t *settings, const uint8_t *buffer)
+{
+    const smartaudioFrameHeader_t *header = (const smartaudioFrameHeader_t *)buffer;
+    const uint8_t fullFrameLength = sizeof(smartaudioFrameHeader_t) + header->length;
+    const uint8_t checkLength = fullFrameLength - 1; // subtract crc byte from length
+    const uint8_t *bufferCrc = buffer + fullFrameLength;
+    uint8_t crc = 0;
+    crc = crc8_dvb_s2_update(crc, buffer, checkLength);
```

Comment:
Simply calculate CRC including incomming crc byte. Result is 0 when CRC is correct